appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -610,12 +646,69 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
    * @returns {number} returns 0 if the blocks are equal, and -1 or 1 if they are not
    */
   function sortByCategory(a,b) {
-    var comparatorA = comparisonName(a).toLowerCase();
-    var comparatorB = comparisonName(b).toLowerCase();
+    // Sort by Category First, also handles other floating blocks
+    if (a.category == b.category && a.category != "Component"){
+      var comparatorA = comparisonName(a).toLowerCase();
```

Comment:
This didn't quite work in the case of variables. I put some top-level floating getter/setter blocks and they got sorted in with the global variables, and not in an obvious way. I'll attach an image and project for reference.